src/test/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretTest.java

```diff
@@ -0,0 +1,186 @@
+package com.iexec.sms.secret.compute;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceException;
+import javax.validation.ConstraintViolationException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@Slf4j
+@DataJpaTest
+public class TeeTaskComputeSecretTest {
+
+    @Autowired
+    private EntityManager entityManager;
+
+    @Autowired
+    private TeeTaskComputeSecretRepository teeTaskComputeSecretRepository;
+
+    private TeeTaskComputeSecret getAppDeveloperSecret() {
+        return TeeTaskComputeSecret.builder()
+                .secretOwnerRole(SecretOwnerRole.APPLICATION_DEVELOPER)
+                .onChainObjectAddress("0x1")
+                .onChainObjectType(OnChainObjectType.APPLICATION)
+                .fixedSecretOwner("")
+                .key("0")
+                .value("secretValue")
+                .build();
+    }
+
+    private TeeTaskComputeSecret getRequesterSecret() {
+        return TeeTaskComputeSecret.builder()
+                .secretOwnerRole(SecretOwnerRole.REQUESTER)
+                .onChainObjectAddress("")
+                .onChainObjectType(OnChainObjectType.APPLICATION)
+                .fixedSecretOwner("0x1")
+                .key("secret-key")
+                .value("secretValue")
+                .build();
+    }
+
+    @BeforeEach
+    void preflight() {
+        teeTaskComputeSecretRepository.deleteAll();
+    }
+
+    @Test
+    void shouldNotSaveSecretFromDefaultBuilder() {
+        log.info("shouldNotSaveEntity");
+        TeeTaskComputeSecret secret = TeeTaskComputeSecret.builder().build();
+        assertThat(secret).hasAllNullFieldsOrProperties();
+        teeTaskComputeSecretRepository.save(secret);
+        assertThatThrownBy(() -> entityManager.flush())
```

Comment:
I think you can remove flushes et add
`@Transactional(propagation = Propagation.NOT_SUPPORTED)`
on top of class.

Reply:
`@Transactional` did not work, I improved by using the `saveAndFlush` method.
See 3113afc

Reply:
Perfect!